datalad/interface/create_collection.py

```diff
@@ -12,17 +12,44 @@
 
 __docformat__ = 'restructuredtext'
 
+
+from os import curdir
+from os.path import join as opj, abspath, expandvars, expanduser
 from .base import Interface
 from datalad.support.param import Parameter
-from datalad.support.constraints import EnsureStr
+from datalad.support.constraints import EnsureStr, EnsureNone
+from datalad.support.collectionrepo import CollectionRepo
+from appdirs import AppDirs
+
+dirs = AppDirs("datalad", "datalad.org")
```

Comment:
Right. And there are/will be more things like that. Just not sure where to
put it best yet.

2015-09-04 19:02 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/interface/create_collection.py
> <https://github.com/datalad/datalad/pull/186#discussion_r38772139>:
>
> >  from .base import Interface
> >  from datalad.support.param import Parameter
> > -from datalad.support.constraints import EnsureStr
> > +from datalad.support.constraints import EnsureStr, EnsureNone
> > +from datalad.support.collectionrepo import CollectionRepo
> > +from appdirs import AppDirs
> > +
> > +dirs = AppDirs("datalad", "datalad.org")
>
> I see this pattern over and over again -- I guess we need some kind of
> 'datalad.dirs' with some helpers like get_user_data_path(filename) which
> would do the magical import, AppDirs, opj?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/186/files#r38772139>.
>